### ISSUE TITLE
Cookies are only sent for the first request

### ISSUE THREAD

**Issue by mc-lep** [rangav/thunder-client-support#164]
**Description**
I have defined a cookie content in the environment file and then in the header of the request I set the cookie header with the environment variable.
When I first click on the send request, everything is OK. The second time i click on the send button the cookie is not sent (the server tells me the cookie is not present).
Then if I restart Visual Studio Code, everything is fine for the first time I click on the send button, but fails the second time.

**To Reproduce**
 - Define an environment variable cookie
 - Defiine a new request
 - Add the header Cookie with the value {{cookie}} 
 - Click the send button
 - Click the send button

**Expected behavior**
The second time I click the send button, the cookie must be sent

**Platform:** 
 - OS: Window 10
 - vscode version: 1.56.1
  - extension version: 1.3.0

**Solution:**
None

**Comment by rangav** [comment]
Hi @mc-lep thanks for reporting, will look into it.

**Comment by rangav** [comment]
HI @mc-lep I am just testing the cookie issue. I can see the value is sent. Can you tell me how to re-produce the issue with url https://httpbin.org/anything
![image](https://user-images.githubusercontent.com/8637550/119222883-6ab28800-baee-11eb-92f2-40d5b57a32b8.png)

**Comment by mercadis** [comment]
Hi @rangav,
the cookie value input is ignored if there was a response with Set-Cookie header.
![image](https://user-images.githubusercontent.com/1176011/119226901-bb84a980-bb0b-11eb-86b0-fc5ac973d9aa.png)
with your actual tool cookies are concatenated:
![image](https://user-images.githubusercontent.com/1176011/119227362-ba547c00-bb0d-11eb-829a-acd7cfc04c56.png)

Thanks for your awesome plugin.

**Comment by f-do** [comment]
Same Issue here, the first time everything is as expected after the first call it is set to an other value.

**Comment by rangav** [comment]
Hi @f-do can you let me know how to reproduce the issue?

**Comment by f-do** [comment]
Yeah for sure: 
![image](https://user-images.githubusercontent.com/4299065/156923847-4ee92ad9-82cc-48da-a6ad-3c09d6117cf2.png)
after the first call i get a set-cookie header with this cookie, after that the extensions don't add more cookies.
![image](https://user-images.githubusercontent.com/4299065/156924005-f8471269-30e3-4cd5-99c0-5ac92cb8b90a.png)

if i clear all then in the first request the cookies from the env or local query is added:
![image](https://user-images.githubusercontent.com/4299065/156923930-32a07728-3677-443e-9035-c7b2b275afcd.png)
after a resend:
![image](https://user-images.githubusercontent.com/4299065/156923956-f00fae19-ce91-4bb7-874a-6a1e5ca94fe3.png)

if i you the generated curl script everything is working as expected:   --header 'Cookie: a=b;c=d;' \ and i is set. 

So basically the Cookie i want so set is not overriding the Cookie what was set.

**Comment by rangav** [comment]
Hi @f-do let me check if I understood right

1. First request you sending with cookie header with value `a=b,c=d`
2. After running the request, the server is sending another cookie in set-cookie header 
3. Now when send request 2nd time you are not seeing the new cookie value from set-cookie earlier received. here you are having issue right?


Please update the steps if anything is missing

**Comment by f-do** [comment]
Hi @rangav,

Yes almost perfectly get the thing:
1. Point the a + c Cookie with the values b and d is arriving as expected
2. Yes after the set cookie that overrides all others
3. correct only the from server responded cookie is set and no Cookie a + c is arrived again. So only the cookie from the server is there.

**Comment by rangav** [comment]
so `a and c` cookie is missing in the 2nd request right?

**Comment by f-do** [comment]
correct, after the 1. Request the a+b is not present anymore.

**Comment by rangav** [comment]
got it thanks, will fix it asap by today or tomorrow.

**Comment by f-do** [comment]
oh thank you very much that will help me a lot ;)

**Comment by rangav** [comment]
Hi all, fixed the bug, please install the pre-release vsix file and test it.

Just want to make sure its working as expected before I publish to everyone

[vscode-thunder-client-1.13.3.vsix.zip](https://github.com/rangav/thunder-client-support/files/8193105/vscode-thunder-client-1.13.3.vsix.zip)

Install manually from extensions tab
![Screenshot 2022-03-05 at 21 34 56](https://user-images.githubusercontent.com/8637550/156900551-8e233bfd-d8a9-4c35-9aac-f8ac999ce708.png)

**Comment by f-do** [comment]
Hi, awesome it works now as expected ;)

**Comment by rangav** [comment]
Thanks @f-do for confirmation

Now its published to **marketplace** as `pre-release` version, so others can also install it.

How to install pre-release version
![Screenshot 2022-03-02 at 15 14 35](https://user-images.githubusercontent.com/8637550/156390587-b6be1a85-92c4-4ba5-9592-62827ae24621.png)

**Comment by rangav** [comment]
This bug is is now fixed and published marketplace as release version in `v1.12.5`

See all features and bugs fixed here
https://github.com/rangav/thunder-client-support/releases/tag/v1.12.5

**Comment by Wiper-R** [comment]
I am still facing same issue even on latest version.

**Comment by rangav** [comment]
Hi @Wiper-R can you share screenshot of how you created cookie and response

**Comment by davinishi** [comment]
@rangav I think this bug is back because I am having the same issue here.

**Comment by rangav** [comment]
@davinishi I just now tested its working for me. can you share any screenshots

**Comment by MasterIfeanyi** [comment]
@ragnav THe bug is still there. 

I have a REST API, when I log-in I get an accessToken and a cookie, but I need to pass the cookie along on `refresh` endpoint to get a refreshToken. 

The cookie is not showing in Thunder Client. 

Watching [this Youtube tutorial](https://www.youtube.com/watch?v=f2EqECiTBL8&t=18366s)

**Comment by rangav** [comment]
@MasterIfeanyi 

you need to save cookie from response to env variable and then use that variable in `refresh` request

docs to save cookie to variable
https://github.com/rangav/thunder-client-support#setenv

please create a new issue, we can discuss there.

**Comment by IfeanyiChima** [comment]
Hi, I need a response @rangav The cookie issue. I don't know how to automatically send the cookie for every request.

**Comment by dsikes** [comment]
I am facing an issue where it sends the old cookie, even if I have specified a new one in the headers. I love Thunder Client! Just trying to figure out how to get the cookie that I have set in the header to be sent with the request instead of the original cookie that I set. Thanks!

**Comment by rangav** [comment]
@dsikes can you please create new issue, with some screenshots of the problem
